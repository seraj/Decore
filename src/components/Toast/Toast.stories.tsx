import { text, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Toast from "./Toast";

const stories = storiesOf("Components/Toast", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <>
      <div>
        <Toast
          type={text("type", "success")}
          text={text("text", "Profile updated!")}
          icon="data:image/svg+xml,%3Csvg width='50' height='50' viewBox='0 0 50 50' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Ccircle cx='25' cy='25' r='25' fill='%233ABF7C'/%3E%3Cpath d='M37.8732 18.6238L35.9116 16.5383C35.8336 16.4461 35.711 16.4 35.5995 16.4C35.477 16.4 35.3655 16.4461 35.2875 16.5383L21.6902 30.699L16.7417 25.5831C16.6526 25.491 16.5411 25.4449 16.4297 25.4449C16.3182 25.4449 16.2068 25.491 16.1176 25.5831L14.1337 27.6341C13.9554 27.8184 13.9554 28.1065 14.1337 28.2908L20.3751 34.7432C20.7763 35.158 21.2667 35.4 21.6791 35.4C22.2698 35.4 22.7825 34.9506 22.9719 34.7663C22.9719 34.7663 22.9719 34.7663 22.9831 34.7663L37.8843 19.2805C38.0404 19.0847 38.0404 18.7966 37.8732 18.6238Z' fill='white'/%3E%3C/svg%3E%0A"
        />

        <br />

        <Toast
          type="info"
          text="You can use this promo code today!!!"
          icon="data:image/svg+xml,%3Csvg width='50' height='50' viewBox='0 0 50 50' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M25 0C11.2019 0 0 11.2019 0 25C0 38.7981 11.2019 50 25 50C38.7981 50 50 38.7981 50 25C50 11.2019 38.7981 0 25 0ZM27.524 37.5H22.476V22.476H27.524V37.5ZM27.524 17.5481H22.476V12.5H27.524V17.5481Z' fill='%234D9DE0'/%3E%3C/svg%3E%0A"
        />

        <br />

        <Toast
          type="warning"
          text="Your balance is low!"
          icon="data:image/svg+xml,%3Csvg width='50' height='44' viewBox='0 0 50 44' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M0 43.1818H50L25 0L0 43.1818ZM27.2727 36.3636H22.7273V31.8182H27.2727V36.3636ZM27.2727 27.2727H22.7273V18.1818H27.2727V27.2727Z' fill='%23FF8552'/%3E%3C/svg%3E%0A"
        />

        <br />
        <Toast
          type="error"
          text="Something went wrong, please try again."
          icon="data:image/svg+xml,%3Csvg width='50' height='50' viewBox='0 0 50 50' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M25 0C11.2 0 0 11.2 0 25C0 38.8 11.2 50 25 50C38.8 50 50 38.8 50 25C50 11.2 38.8 0 25 0ZM27.5 37.5H22.5V32.5H27.5V37.5ZM27.5 27.5H22.5V12.5H27.5V27.5Z' fill='%23DB504A'/%3E%3C/svg%3E%0A"
        />
      </div>
    </>
  ))
);
