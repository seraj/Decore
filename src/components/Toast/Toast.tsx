import * as React from "react";
import StyledToast, {
  ToastIconContainer,
  ToastContainer,
  ToastContent
} from "./styled/StyledToast";
import Icon from "../Icon";
import ToastProps from "./Toast.props";

const Toast: React.FC<ToastProps> = ({
  closeToast,
  icon = "",
  type,
  text,
  ...props
}) => (
  <ToastContainer type={type} {...props}>
    <StyledToast type={type}>
      <ToastIconContainer>
        <Icon src={icon} />
      </ToastIconContainer>
      <ToastContent>
        <h3>{type}</h3>
        <p>{text}</p>
      </ToastContent>
    </StyledToast>
    <Icon
      style={{ cursor: "pointer" }}
      onClick={closeToast}
      src="data:image/svg+xml,%3Csvg width='16' height='16' viewBox='0 0 16 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M16 1.61714L14.3829 0L8 6.38286L1.61714 0L0 1.61714L6.38286 8L0 14.3829L1.61714 16L8 9.61714L14.3829 16L16 14.3829L9.61714 8L16 1.61714Z' fill='%23999999'/%3E%3C/svg%3E%0A"
    />
  </ToastContainer>
);

export default Toast;
