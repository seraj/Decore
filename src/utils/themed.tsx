import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
