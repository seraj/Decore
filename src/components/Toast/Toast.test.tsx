import * as React from "react";

import themed from "../../utils/themed";
import Toast from "./Toast";

describe("Toast", () => {
  it("should render correctly", () => {
    const component = themed(<Toast>Test</Toast>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
