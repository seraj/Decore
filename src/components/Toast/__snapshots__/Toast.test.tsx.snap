// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 550px;
  padding: 0.5em 1em;
  background: #ffffff;
  box-shadow: 1px 1px 5px rgba(0,0,0,0.05);
  border-left: 4px solid #3abf7c;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  justify-self: flex-end;
  position: relative;
  margin: 5px 0;
  top: 5px;
}

.c5 h3 {
  font-weight: 600;
  font-size: 18px;
  text-transform: capitalize;
}

.c5 p {
  font-weight: 500;
  font-size: 16px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: right;
  -webkit-justify-content: right;
  -ms-flex-pack: right;
  justify-content: right;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c2 span {
  width: auto;
}

.c4 {
  display: inline-block;
  fill: #333333;
  width: 12px;
}

<div
  className="c0"
  type="success"
>
  <div
    className="c1"
    type="success"
  >
    <div
      className="c2 c3"
    >
      <span
        className="isvg loaded c4"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg width='50' height='50' viewBox='0 0 50 50' fill='none' xmlns='http://www.w3.org/2000/svg'><circle cx='25' cy='25' r='25' fill='#3ABF7C'/><path d='M37.8732 18.6238L35.9116 16.5383C35.8336 16.4461 35.711 16.4 35.5995 16.4C35.477 16.4 35.3655 16.4461 35.2875 16.5383L21.6902 30.699L16.7417 25.5831C16.6526 25.491 16.5411 25.4449 16.4297 25.4449C16.3182 25.4449 16.2068 25.491 16.1176 25.5831L14.1337 27.6341C13.9554 27.8184 13.9554 28.1065 14.1337 28.2908L20.3751 34.7432C20.7763 35.158 21.2667 35.4 21.6791 35.4C22.2698 35.4 22.7825 34.9506 22.9719 34.7663C22.9719 34.7663 22.9719 34.7663 22.9831 34.7663L37.8843 19.2805C38.0404 19.0847 38.0404 18.7966 37.8732 18.6238Z' fill='white'/></svg>
",
          }
        }
      />
    </div>
    <div
      className="c5"
    >
      <h3>
        success
      </h3>
      <p>
        Profile Updated.
      </p>
    </div>
  </div>
  <span
    className="isvg loaded c4"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg width='16' height='16' viewBox='0 0 16 16' fill='none' xmlns='http://www.w3.org/2000/svg'><path d='M16 1.61714L14.3829 0L8 6.38286L1.61714 0L0 1.61714L6.38286 8L0 14.3829L1.61714 16L8 9.61714L14.3829 16L16 14.3829L9.61714 8L16 1.61714Z' fill='#999999'/></svg>
",
      }
    }
    style={
      Object {
        "cursor": "pointer",
      }
    }
  />
</div>
`;
